akka-javasdk/src/main/java/akka/javasdk/workflow/Workflow.java

```diff
@@ -489,18 +510,226 @@ public Optional<MaxRetries> getFailoverMaxRetries() {
     }
   }
 
+  public sealed interface Settings {
 
-  public WorkflowDef<S> workflow() {
-    return new WorkflowDef<>();
+
+    Optional<Duration> workflowTimeout();
+
+    Optional<Duration> defaultStepTimeout();
+
+    Optional<RecoverStrategy<?>> stepRecoverStrategy();
+
+    Optional<String> failoverStepName();
+
+    Optional<?> failoverStepInput();
+
+    Optional<MaxRetries> failoverMaxRetries();
+
+    List<StepConfig> stepConfigs();
+
+    Optional<Step> findByName(String name);
   }
 
+  private record LegacySettings<S>(WorkflowDef<S> legacyDefinition) implements Settings {
+
+    @Override
+    public Optional<Duration> workflowTimeout() {
+      return legacyDefinition.getWorkflowTimeout();
+    }
+
+    @Override
+    public Optional<RecoverStrategy<?>> stepRecoverStrategy() {
+      return legacyDefinition.getStepRecoverStrategy();
+    }
+
+    @Override
+    public Optional<String> failoverStepName() {
+      return legacyDefinition.getFailoverStepName();
+    }
+
+    @Override
+    public Optional<?> failoverStepInput() {
+      return legacyDefinition.getFailoverStepInput();
+    }
+
+    @Override
+    public Optional<MaxRetries> failoverMaxRetries() {
+      return legacyDefinition.getFailoverMaxRetries();
+    }
+
+    @Override
+    public Optional<Duration> defaultStepTimeout() {
+      return legacyDefinition.getStepTimeout();
+    }
+
+    @Override
+    public List<StepConfig> stepConfigs() {
+      return legacyDefinition.getStepConfigs();
+    }
+
+    @Override
+    public Optional<Step> findByName(String name) {
+      return legacyDefinition.findByName(name);
+    }
+  }
+
+  private record SettingsImpl(
+      Workflow<?> instance,
+      List<StepMethod> stepMethods,
+      List<StepConfig> stepConfigs,
```

Comment:
This is not yet in its best shape. It's awkward that Settings has the instance and the methods.

I want to make this evolve, but we need to move in small steps.